packages/nimbus/src/components/date-range-picker/date-range-picker.dev.mdx

```diff
@@ -0,0 +1,407 @@
+    render(<DateRangePicker />);
+
+    const calendarButton = screen.getByRole('button', { name: /open calendar/i });
+    await user.click(calendarButton);
+
+    await waitFor(() => {
+      expect(screen.getByRole('grid')).toBeInTheDocument();
+    });
+  });
+
+  it('clears selection when clear button is clicked', async () => {
+    const user = userEvent.setup();
+    const handleChange = jest.fn();
+    render(<DateRangePicker onChange={handleChange} />);
+
+    const clearButton = screen.getByRole('button', { name: /clear/i });
+    await user.click(clearButton);
+
+    expect(handleChange).toHaveBeenCalledWith(null);
+  });
+});
+```
+
+### Testing with date values
+
+Test date value handling using `@internationalized/date`:
+
+```tsx
+import { render, screen } from '@testing-library/react';
+import { CalendarDate } from '@internationalized/date';
+import { DateRangePicker } from '@commercetools/nimbus';
+
+describe('DateRangePicker with date values', () => {
+  it('displays provided date range', () => {
+    const dateRange = {
+      start: new CalendarDate(2024, 1, 15),
+      end: new CalendarDate(2024, 1, 20),
+    };
+
+    render(<DateRangePicker value={dateRange} />);
+
+    // Verify the dates are displayed in the input segments
+    expect(screen.getByText('1', { selector: '[role="spinbutton"]' })).toBeInTheDocument(); // Month
+    expect(screen.getByText('15', { selector: '[role="spinbutton"]' })).toBeInTheDocument(); // Start day
+    expect(screen.getByText('20', { selector: '[role="spinbutton"]' })).toBeInTheDocument(); // End day
+  });
+
+  it('calls onChange with correct date types', async () => {
+    const user = userEvent.setup();
+    const handleChange = jest.fn();
+    render(<DateRangePicker onChange={handleChange} />);
+
+    // Interact with date picker...
+    // After selection:
+    expect(handleChange).toHaveBeenCalledWith(
+      expect.objectContaining({
+        start: expect.any(CalendarDate),
+        end: expect.any(CalendarDate),
+      })
+    );
+  });
+});
+```
+
+### Testing time selection
+
+When using time granularity, test time input behavior:
+
+```tsx
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { DateRangePicker } from '@commercetools/nimbus';
+
+describe('DateRangePicker with time', () => {
+  it('shows time inputs when granularity includes time', async () => {
+    const user = userEvent.setup();
+    render(<DateRangePicker granularity="minute" />);
+
+    // Open calendar
+    const calendarButton = screen.getByRole('button', { name: /open calendar/i });
+    await user.click(calendarButton);
+
+    // Time inputs should appear in the popover footer
+    await waitFor(() => {
+      expect(screen.getByText(/start time/i)).toBeInTheDocument();
+      expect(screen.getByText(/end time/i)).toBeInTheDocument();
+    });
+  });
+});
+```
+
+## API reference
+
+<PropsTable id="DateRangePicker" />
```

Comment:
I think the Props table should come before the testing section. My care amount here is relatively high.